casbin/enforcer.py

```diff
@@ -174,19 +174,19 @@ def get_implicit_users_for_permission(self, *permission):
         get_implicit_users_for_permission("data1", "read") will get: ["alice", "bob"].
         Note: only users will be returned, roles (2nd arg in "g") will be excluded.
         """
-        subjects = self.get_all_subjects()
-        roles = self.get_all_roles()
-
-        users = set_subtract(subjects, roles)
+        p_subjects = self.get_all_subjects()
+        g_inherit = self.model.get_values_for_field_in_policy("g", "g", 1)
+        g_subjects = self.model.get_values_for_field_in_policy("g", "g", 0)
+        subjects = array_remove_duplicates(p_subjects + g_subjects)
 
         res = list()
-        for user in users:
+        for user in subjects:
             req = join_slice(user, *permission)
             allowed = self.enforce(*req)
 
             if allowed:
                 res.append(user)
-
+        res = set_subtract(res, g_inherit)
```

Comment:
Modify this line to: 
```
subjects = set_subtract(subjects, g_inherit)
```
and put it before the `for` loop to improve performance .

Reply:
sure that makes sense